packages/shared_preferences/lib/shared_preferences_tizen.dart

```diff
@@ -82,9 +83,34 @@ class SharedPreferencesPlugin extends SharedPreferencesStorePlatform {
     return tizen.preference_remove_all() == 0;
   }
 
+  @override
+  Future<bool> clearWithParameters(ClearParameters parameters) async {
+    final PreferencesFilter filter = parameters.filter;
+    final List<String> keys = List<String>.of(_preferences.keys);
+    bool failed = false;
+    for (final String key in keys) {
+      if (key.startsWith(filter.prefix) &&
+          (filter.allowList == null || filter.allowList!.contains(key))) {
+        failed |= !(await remove(key));
```

Comment:
`!(await remove(key))` This code returns false even if remove succeeds.

Reply:
About integration_test, `getAllWithParameters` and `clearWithParameters` are used by other methods in original shared_preferences package. It seems they omit test about two api's because if they don't work properly, other methods won't work properly.

Reply:
Fix by returning `!failed` at last of this method. This makes return `false` when `remove` of any key failed.

Reply:
OK but I don't know if we should stop or continue the for loop if remove fails. What do you think?
The Linux implementation writes `preference` and returns false if an exception occurs. when an exception occurs, further writing stops.

(If you will the for loop to stop when remove fails, you can replace `failed` with `ret` (like other code).)

Reply:
Then it would be good to stop on exception like other implementations for unity. I'll fix it.

Reply:
Fixed to return false right after remove fail.